/* This file is part of Clementine.
   Copyright 2011, David Sansome <me@davidsansome.com>

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

// Note: this file is licensed under the Apache License instead of GPL because
// it is used by the Spotify blob which links against libspotify and is not GPL
// compatible.


package protobuf;

message LoginRequest {
  required string username = 1;
  required string password = 2;
}

message LoginResponse {
  required bool success = 1;
  required string error = 2;
}

message Playlists {
  message Playlist {
    required int32 index = 1;
    required string name = 2;
  }

  repeated Playlist playlist = 1;
}

message SpotifyMessage {
  optional LoginRequest login_request = 1;
  optional LoginResponse login_response = 2;
  optional Playlists playlists_updated = 3;
}
